python-frontend/src/main/java/org/sonar/python/tree/CallExpressionImpl.java

```diff
@@ -188,6 +190,21 @@ public PythonType typeV2() {
     if (callee().typeV2() instanceof ClassType classType) {
       return new ObjectType(classType);
     }
+    if (callee().typeV2() instanceof FunctionType functionType) {
+      return functionType.returnType();
+    }
+    if (callee().typeV2() instanceof UnionType unionType) {
+      PythonType result = null;
+      for (PythonType candidate : unionType.candidates()) {
+        if (candidate instanceof ClassType classType) {
+          result = result == null ? classType : UnionType.or(result, classType);
+        }
+        if (candidate instanceof FunctionType functionType) {
+          result = result == null ? functionType.returnType() : UnionType.or(result, functionType.returnType());
```

Comment:
maybe it makes sense to have this logic as a part of the `UnionType.or` method if the first argument is null - return the second one